pkg/virt-controller/watch/job.go

```diff
@@ -15,24 +16,27 @@ import (
 )
 
 func migrationJobSelector() kubeapi.ListOptions {
-	fieldSelector := fields.Everything()
-	labelSelector, err := labels.Parse(kvirtv1.DomainLabel)
+	fieldSelector := fields.ParseSelectorOrDie(
+		"status.phase!=" + string(v1.PodPending) +
+			",status.phase!=" + string(v1.PodRunning) +
+			",status.phase!=" + string(v1.PodUnknown))
+	labelSelector, err := labels.Parse(kvirtv1.DomainLabel + "," + kvirtv1.MigrationLabel)
 	if err != nil {
 		panic(err)
 	}
 	return kubeapi.ListOptions{FieldSelector: fieldSelector, LabelSelector: labelSelector}
 }
 
-func NewJobController(vmService services.VMService, recorder record.EventRecorder, restClient *rest.RESTClient) (cache.Store, *kubecli.Controller) {
+func NewJobController(vmService services.VMService, recorder record.EventRecorder, clientSet *kubernetes.Clientset, restClient *rest.RESTClient) (cache.Store, *kubecli.Controller) {
 	selector := migrationJobSelector()
-	lw := kubecli.NewListWatchFromClient(restClient, "jobs", kubeapi.NamespaceDefault, selector.FieldSelector, selector.LabelSelector)
+	lw := kubecli.NewListWatchFromClient(clientSet.CoreV1().RESTClient(), "pods", kubeapi.NamespaceDefault, selector.FieldSelector, selector.LabelSelector)
```

Comment:
Pretty sure. ListWatchers require a rest client. Using our TPR rest client does not make sense, since pods are built in. If we do queries we should use the core client on built in types, but sdk infra components like ListWatchers are built arund the rest client.